crates/utils/src/utils.rs

```diff
@@ -116,26 +115,28 @@ pub fn scrape_text_for_mentions(text: &str) -> Vec<MentionData> {
   out.into_iter().unique().collect()
 }
 
-pub fn is_valid_username(name: &str) -> bool {
-  VALID_USERNAME_REGEX.is_match(name)
+pub fn is_valid_actor_name(name: &str) -> bool {
+  let max_length = Settings::get()
+    .actor_name_max_length
+    .unwrap_or_else(|| Settings::default().actor_name_max_length.unwrap());
+  name.chars().count() <= max_length && VALID_ACTOR_NAME_REGEX.is_match(name)
```

Comment:
What I mean is just move the regex in here, and have the length check be in the regex. No sense in doing two checks when we can just do one regex check.

Reply:
Then the regex needs to be compiled on every call. I suppose that could even be used for a ddos.